sql/core/src/test/scala/org/apache/spark/sql/json/JsonSuite.scala

```diff
@@ -1074,4 +1073,16 @@ class JsonSuite extends QueryTest {
     assert(StructType(Seq()) === emptySchema)
   }
 
+  test("SPARK-7565 MapType in JsonRDD") {
+    val schemaWithSimpleMap = StructType(
+      StructField("map", MapType(StringType, IntegerType, true), false) :: Nil)
+    val df = read.schema(schemaWithSimpleMap).json(mapType1)
+    val temp = Utils.createTempDir()
+    df.write.mode("overwrite").parquet(temp.getPath)
+
+    setConf("spark.sql.json.useJacksonStreamingAPI", "false")
```

Comment:
Since we are using a singleton `TestSQLContext`, this setting will also affect other tests. How about we get the old value at the beginning of the test and set the old value back at the end of the test? Also, let's also use `checkAnswer` to make sure the data we read form the parquet data is correct.

btw, it will be good to test corrupt record column as well. We can have a bad JSON record and then ask Spark SQL to infer the schema. Then, in the schema, there will be a column called `_corrupt_record`.